queenbee/io/reference.py

```diff
@@ -0,0 +1,256 @@
+"""Objects to reference parameters, files and folders from inputs, tasks and items."""
+
+import re
+from typing import List, Union
+from pydantic import Field, constr
+
+from ..base.basemodel import BaseModel
+
+
+def template_string(keys: List[str]) -> str:
+    """Generate a template string from a list of key
+
+    Arguments:
+        keys {List[str]} -- A list of keys
+
+    Returns:
+        str -- A template string
+    """
+    return f'{{{{{".".join(keys)}}}}}'
+
+
+class _BaseReference(BaseModel):
+    """A Base reference model."""
+
+    type: constr(regex='^_BaseReference$') = '_BaseReference'
+
+    @property
+    def source(self):
+        return None
+
+    def fetch_from_flow(self, flow):
+        pass
+
+
+class FileReference(_BaseReference):
```

Comment:
Are these "Reference" only used inside the tasks? Won't be used as a input type for recipe, right?

Reply:
Yes. These are just reference objects which are similar to wires in Grasshopper. It says from where should I source the values for an input.

Reply:
Here is a supporting graphic.

![inputs_outpus](https://user-images.githubusercontent.com/2915573/95382095-1fc56680-08b7-11eb-89bc-9913e498832e.png)